spec/apply_spec.rb

```diff
@@ -54,6 +54,25 @@
         output = bora.run(bora_config, 'apply', 'web-prod', expect_exception: true)
         expect(output).to include(format(Bora::Stack::STACK_ACTION_FAILURE_MESSAGE, 'Update', 'web-prod'))
       end
+
+      it 'removes create stack only api parameters when updating a stack' do
+        stack_config = {
+          'on_failure' => 'DELETE',
+          'capabilities' => ['CAPABILITY_IAM']
+        }
+
+        bora_config.templates.web.stacks.prod.merge!(stack_config)
+        expect(stack).to receive(:update)
+          .with(
+            hash_including(
+              :template_body,
+              'capabilities' => ['CAPABILITY_IAM']
+            )
+          )
+          .and_return(true)
```

Comment:
There's couple of problems here. The first is that this spec is probably not testing what you think it is... which is my fault for making things confusing as hell :)  Let me try to explain.

So there's two `stack.rb` files - `bora/stack.rb` and `bora/cfn/stack.rb` - I know, confusing and I should rename the latter to `cfn_stack.rb`. The change you made in this PR was in `bora/cfn/stack.rb`.

Now this particular spec is testing `bora/stack.rb`. The way it does that is by mocking out `bora/cfn/stack.rb` and asserting that it gets called the right way. So in effect the "stack" variable here is an instance of `Bora::Cfn::Stack`. The reason I did it this way is that mocking out the AWS API is a pain, so I decided to confine all that in `bora/cfn/stack.rb` and test it in isolation, and then for everything else just test that it calls it the right way.

So long story short, this test needs to be in `spec/cfn/stack_spec.rb` (around line 147). If you're not comfortable with that I'll happily implement the test myself.

The other problem (if we assume the first problem didn't exist) is that this test should probably explicitly check that certain hash keys were _not_ present using `hash_not_including`.